pkg/driver/controller.go

```diff
@@ -59,7 +72,40 @@ const (
 	volumeParamsExtraTags                     = "extraTags"
 )
 
-func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest) (*csi.CreateVolumeResponse, error) {
+// controllerService represents the controller service of CSI driver
+type controllerService struct {
+	cloud         cloud.Cloud
+	inFlight      *internal.InFlight
+	driverOptions *DriverOptions
+}
+
+// newControllerService creates a new controller service
+// it panics if failed to create the service
+func newControllerService(driverOptions *DriverOptions) controllerService {
+	region := os.Getenv("AWS_REGION")
+	if region == "" {
+		klog.V(5).InfoS("[Debug] Retrieving region from metadata service")
+		metadata, err := cloud.NewMetadataService(cloud.DefaultEC2MetadataClient, cloud.DefaultKubernetesAPIClient, region)
+		if err != nil {
+			klog.ErrorS(err, "Could not determine region from any metadata service. The region can be manually supplied via the AWS_REGION environment variable.")
```

Comment:
missing `klog.FlushAndExit(klog.ExitFlushTimeout, 1)`

Reply:
this is currently the same behavior as EBS: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/pkg/driver/controller.go#L88-L89 Should a similar change be made to the ebs csi driver?

Reply:
panic also works here, I don't feel strongly about it.

Reply:
I'm going to add a note about this for more research, it's possible we should just move all panics to this